src/meta/meta_ckpt.c

```diff
@@ -1064,6 +1071,14 @@ __wt_meta_ckptlist_to_meta(WT_SESSION_IMPL *session, WT_CKPT *ckptbase, WT_ITEM
           (int64_t)ckpt->ta.newest_stop_durable_ts, (int64_t)ckpt->ta.newest_stop_ts,
           (int64_t)ckpt->ta.newest_stop_txn, (int)ckpt->ta.prepare, (int64_t)ckpt->write_gen,
           (int64_t)ckpt->run_write_gen));
+
+        /*
+         * Update the Max connection write generation number. We exclude the metadata write
+         * generation informations as this information is going to be saved in the metadata file
+         * itself for every checkpoint.
+         */
+        if (session->dhandle == NULL || !WT_IS_METADATA(session->dhandle))
+            S2C(session)->write_gen = WT_MAX(ckpt->write_gen, S2C(session)->write_gen);
```

Comment:
The algorithm of updating connection `write_gen` is duplicated in the line above and `__wt_metadata_update_base_write_gen`. It would be better if we extract this logic into a separate method. Just in case we want to change the algorithm we wouldn't have to change it in two places. It would be safer.